libcontainer/container_linux.go

```diff
@@ -169,7 +169,17 @@ func (c *linuxContainer) OCIState() (*specs.State, error) {
 }
 
 func (c *linuxContainer) Processes() ([]int, error) {
-	pids, err := c.cgroupManager.GetAllPids()
+	var pids []int
+	status, err := c.currentStatus()
+	if err != nil {
+		return pids, err
+	}
+	// for systemd cgroup, the unit's cgroup path will be auto removed if container's all processes exited
+	if status == Stopped && !c.cgroupManager.Exists() {
```

Comment:
Does it make sense to check for "Created" here as well? Or should it return an error in such case?

Reply:
No it does not ("created" container has a `runc init` process running so everything works as expected.

Also, "paused" container also has a process running.

So, it seems that the check for "stopped" here is sufficient.